cranelift/isle/isle/src/lexer.rs

```diff
@@ -88,6 +88,17 @@ impl<'a> Lexer<'a> {
 
     /// Create a new lexer from the given files.
     pub fn from_files<P>(file_paths: impl IntoIterator<Item = P>) -> Result<Lexer<'a>>
+    where
+        P: AsRef<Path>,
+    {
+        Self::from_files_read(file_paths, std::fs::read_to_string)
+    }
+
+    /// Create a new lexer from the given files.
+    pub fn from_files_read<P>(
+        file_paths: impl IntoIterator<Item = P>,
+        mut read: impl FnMut(PathBuf) -> std::io::Result<String>,
```

Comment:
Rather than the callback approach, maybe it would be simpler to just provide a list of paths and strings? Something like:

`pub fn from_file_contents(files: Vec<(PathBuf, String)>) -> Result<Lexer<'a>>` ?

Reply:
@cfallin   Yeah,sure.